.travis.yml

```diff
@@ -1,28 +1,65 @@
 language: scala
+
 scala:
    - 2.11.12
+
 jdk:
  - oraclejdk8
 
-# These directories are cached to S3 at the end of the build
-cache:
-  directories:
-    - $HOME/.ivy2/cache
-    - $HOME/.sbt/boot/
+sudo: required
+
+services:
+  - docker
+
+addons:
+  chrome: stable
+
+env:
+  global:
+    - APPLICATION_NAME=nussknacker
+    - SANITIZED_BRANCH=`echo ${TRAVIS_BRANCH} | sed 's/[^a-zA-Z0-9]/\_/g' | awk '{print tolower($0)}'`
+    - BUILD_VERSION=$(if [[ -n $TRAVIS_TAG ]]; then echo "${SANITIZED_BRANCH}"; else echo "${SANITIZED_BRANCH}-$TRAVIS_COMMIT"; fi)
+
+before_install:
+  - if [[ "$TRAVIS_PULL_REQUEST" == "false" ]]; then echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin; fi
 
 before_cache:
   # Cleanup the cached directories to avoid unnecessary cache updates
   - rm -fv $HOME/.ivy2/.sbt.ivy.lock
   - find $HOME/.ivy2/cache -name "ivydata-*.properties" -print -delete
   - find $HOME/.sbt        -name "*.lock"               -print -delete
 
-script:
- - sbt ++$TRAVIS_SCALA_VERSION clean test management/it:test
-
-sudo: required
+# These directories are cached to S3 at the end of the build
+cache:
+  directories:
+    - $HOME/.ivy2/cache
+    - $HOME/.sbt/boot/
 
-services:
-  - docker
+stages:
```

Comment:
Please add here a TODO comment that we should use artifacts deployed on S3 between steps (https://docs.travis-ci.com/user/build-stages/#data-persistence-between-stages-and-jobs) or we should move to circle-ci.

Reply:
Ok.